ruby/private/toolchains/ruby_runtime.bzl

```diff
@@ -1,6 +1,21 @@
-load(":bundler.bzl", "install_bundler")
 load("//ruby/private:constants.bzl", "RULES_RUBY_WORKSPACE_NAME")
-load("//ruby/private/tools:repository_context.bzl", "ruby_repository_context")
+load("//ruby/private/toolchains:repository_context.bzl", "ruby_repository_context")
+
+def _install_ruby_version(ctx, version):
+    print("download and extract ruby-build")
+    ctx.download_and_extract(
+        url = "https://github.com/rbenv/ruby-build/archive/v20191205.tar.gz",
+        sha256 = "d8ffe806a215b3afacead72e766f293ce380c78a143911b84cdb5f33e20a5284",
+        stripPrefix = "ruby-build-20191205",
+    )
+
+    install_path = "./build"
+    print("./bin/ruby-build", "--verbose", version, install_path)
+    ctx.execute(
+        ["./bin/ruby-build", "--verbose", version, install_path],
```

Comment:
This is good enough to start with. But could you let me know what was wrong with #20?

Reply:
@yugui My arrival at this PR was:

1. First I tried to use sorbets ruby BUILD file, which I got mostly working, there was one main issue with that approach, the bazel BUILD was executed after repository download (e.g. bundler) so then we would have to reimplement all scripts in `repository_rules` to be python or something, including bundler.
2. Next I tried building ruby directly from source. I also got this working, but there are load of complications, especially openssl integration requires downloading and compiling. I gave up on this when I realized the amount of complexity in building ruby.
3. `ruby-build` removes a lot of that complexity by setting up all the args, and fixes other problems like finding and/or building libraries like zlib and openssl and if it can't find them. The fact that this project exists and is so popular demonstrates the complexity in building ruby. 
 
Ultimately, the main issue with ruby is that there is no good distribution of pre-built ruby binaries for us to just download and use. This might be the ultimate solution to try and build "blessed" ruby versions to be distributed. 

This solution has some nice value, a user can now build against any ruby version without having to install a tool like rbenv. But if the local version installed is good then the rules can just use that without having to build a new version.